### PR TITLE
feat: collection view & autocompletion UX improvements (Step 4.6)

### PR DESCRIPTION
**Pull request by tnaum-ms**
## Step 4.6: Collection View & Autocompletion UX Improvements

Post-review UX fixes and enhancements for the autocompletion system built in Steps 4–4.5.

#### Completion Provider Fixes
- **Documentation links clickable** — Set `isTrusted: true` on completion documentation MarkdownStrings so `[DocumentDB Docs](https://...)` links render as clickable hyperlinks
- **Field completion data preserved across queries** — Removed `SchemaAnalyzer.reset()` on query change so field knowledge accumulates monotonically; queries returning 0 results no longer wipe the field list
- **`$not` moved to field-level** — Removed `$not` from `KEY_POSITION_OPERATORS`; it now correctly appears at operator/value position (`{ price: { $not: { $gt: 1.99 } } }`) instead of key position where it is invalid
- **Project editor value completions** — At value position in the project editor, show `1` (include) and `0` (exclude) instead of filter-specific completions
- **Sort editor value completions** — At value position in the sort editor, show `1` (ascending) and `-1` (descending)
- **Top-level wrapping** — When the editor has no braces (user cleared content), field completions wrap with `{ fieldName: $1 }` and operator snippets keep their full brace-wrapping
- **Auto-trigger characters** — Added `:`, `,`, `[` to trigger characters with string-literal detection (`isCursorInsideString`) to suppress completions inside strings
- **Smart-trigger after `: ` and `, `** — After selecting a field completion, typing space triggers the suggestion popup via `editor.action.triggerSuggest` (50ms delay)
- **Empty editor = key-position** — Split EMPTY from UNKNOWN context: empty editor (no braces) now shows key-position completions (fields + root operators) with `{ }` wrapping, instead of showing all operators. UNKNOWN remains as full discovery fallback.
- **Standalone flag for operators** — Added `standalone?: boolean` to `OperatorEntry`. Non-standalone operators (geospatial sub-operators, `$`, `$natural`) are excluded from completion lists but remain in the registry for hover documentation.

#### Hover Provider Enhancements
- **Field type hover** — Extended `HoverProvider` to show field type info (bold field name, sparse indicator, "Inferred Types" section with all observed types for polymorphic fields)
- **Quoted string key hover** — Added `extractQuotedKey` helper so hover works on `"address.street"` key names (Monaco's `getWordAtPosition` treats `.` and `"` as word boundaries)
- **Hover links actionable** — Added delegated click handler routing Monaco hover `<a>` clicks through `trpcClient.common.openUrl.mutate()` → `vscode.env.openExternal` (webview CSP blocks direct navigation)
- **Polymorphic types** — Threaded `bsonTypes`/`displayTypes` through `FieldCompletionData` from `FieldEntry` for multi-type field display
- **Hover formatting cleanup** — Removed unnecessary line breaks in hover documentation links, enhanced inferred types section formatting

#### Accessibility Fixes
- **ESC respects suggest widget / snippet mode** — ESC now only exits the editor when the suggest widget and snippet mode are both inactive. First ESC dismisses the autocomplete popup, another ESC exits snippet mode, and only then does ESC move focus out.
- **Tab respects snippet tab stops** — Tab no longer moves focus out of the editor during snippet tab-stop navigation. Uses `addAction` with `precondition: '!inSnippetMode'` so Monaco's built-in snippet Tab handler takes over during snippet sessions.
- **Ghost selection fix** — Snippet tab-stop highlights no longer persist and expand after accepting a value completion. Added `cancelSnippetSession()` via `snippetController2.cancel()`: cancels on editor blur and when delimiter characters (`,`, `}`, `]`) are typed.

#### Testing
- **TDD behavior contract tests** — 26 category-based TDD tests verifying completion behavior at each cursor position (EMPTY, KEY, VALUE, OPERATOR, ARRAY-ELEMENT, UNKNOWN). All passing.
- **`isCursorInsideString` tests** — 14 edge-case tests for string-literal detection
- **`extractQuotedKey` tests** — Edge-case coverage for quoted key extraction

### Files Changed
- `src/webviews/documentdbQuery/completions/` — completion knowledge, item creation, mapping
- `src/webviews/documentdbQuery/registerLanguage.ts` — trigger chars, hover provider, quoted key extraction
- `src/webviews/documentdbQuery/documentdbQueryHoverProvider.ts` — field hover, isTrusted, format
- `src/webviews/documentdbQuery/isCursorInsideString.ts` — new module for string detection
- `src/webviews/documentdb/collectionView/components/queryEditor/QueryEditor.tsx` — smart-trigger, link handler
- `src/webviews/components/MonacoEditor.tsx` — ESC context precondition for suggest/snippet mode
- `src/webviews/components/MonacoAutoHeight.tsx` — Tab respects snippet mode via addAction
- `src/documentdb/ClusterSession.ts` — removed schema reset on query change
- `src/utils/json/data-api/autocomplete/toFieldCompletionItems.ts` — bsonTypes/displayTypes threading
- `packages/documentdb-constants/` — standalone flag, int32 BSON type fix, operator overrides
- `src/webviews/documentdbQuery/tdd/` — TDD behavior contract tests
- Test files for all of the above

### CODE REVIEW

File: src/webviews/documentdbQuery/documentdbQueryHoverProvider.ts [hunk 1]
```diff
@@ -35,16 +46,57 @@ export function getHoverContent(word: string): monacoEditor.languages.Hover | nu
...
+    if (typeList && typeList.length > 0) {
+        lines.push('---');
+        lines.push('<br>');
+        lines.push(`Inferred Type: ${typeList.map((type) => `\`${type}\``).join(', ')}`);
+    }
+
+    return {
+        contents: [{ value: lines.join('\n\n'), isTrusted: true, supportHtml: true }],
+    };
+}
```
[the remark this answers is not in the document]
Comment (tnaum-ms) [reply src/webviews/documentdbQuery/documentdbQueryHoverProvider.ts:104]: Fixed. Field hovers no longer set `isTrusted`, and all field-derived strings (`fieldName`, display types) are escaped via a new `escapeMarkdown` utility before interpolation. `supportHtml` is kept for formatting (`<small>`, `&nbsp;`); `isTrusted` is preserved only for operator hovers which render extension-authored content with doc links.

See commit 4a44ff5.

File: src/webviews/documentdbQuery/registerLanguage.ts [hunk 2]
```diff
@@ -163,3 +212,91 @@ async function doRegisterLanguage(monaco: typeof monacoEditor): Promise<void> {
...
+ * along with the 0-based start/end positions of the full quoted string
+ * (including the quotes themselves, for hover range highlighting).
+ *
+ * Returns null if the cursor is not inside a quoted string.
+ *
+ * @param line - the full line content
+ * @param col0 - 0-based column position of the cursor
+ */
+export function extractQuotedKey(line: string, col0: number): { key: string; start: number; end: number } | null {
+    if (col0 < 0 || col0 >= line.length) return null;
```
[the remark this answers is not in the document]
Comment (tnaum-ms) [reply src/webviews/documentdbQuery/registerLanguage.ts]: Done. Moved `extractQuotedKey` and `tryMatchAsClosingQuote` into a dedicated `extractQuotedKey.ts` module. The test now imports from there directly, removing the transitive dependency on Monaco registration wiring.

See commit 91b08d4.

File: src/webviews/documentdb/collectionView/components/queryEditor/QueryEditor.tsx [hunk 3]
```diff
@@ -131,11 +131,90 @@ export const QueryEditor = ({ onExecuteRequest }: QueryEditorProps): JSX.Element
...
+     * Also cancels any active snippet session when a delimiter character
+     * (`,`, `}`, `]`) is typed, preventing the "ghost selection" bug
+     * where the tab-stop highlight expands as the user continues typing.
+     *
+     * Returns a cleanup function.
+     */
+    const setupSmartTrigger = (editor: monacoEditor.editor.IStandaloneCodeEditor): (() => void) => {
+        let triggerTimeout: ReturnType<typeof setTimeout>;
+        const contentDisposable = editor.onDidChangeModelContent((e) => {
+            clearTimeout(triggerTimeout);
```
[the remark this answers is not in the document]
Comment (tnaum-ms) [reply src/webviews/documentdb/collectionView/components/queryEditor/QueryEditor.tsx:166]: Won't fix. `clearTimeout(undefined)` is a valid no-op in JS/TS — the same pattern is used in `setupValidation` a few lines above. There is no TS2454 error; the project compiles cleanly.

File: src/webviews/documentdb/collectionView/components/queryEditor/QueryEditor.tsx [hunk 4]
```diff
@@ -131,11 +131,90 @@ export const QueryEditor = ({ onExecuteRequest }: QueryEditorProps): JSX.Element
...
+            // contain commas or braces as part of the snippet text.
+            if (change.text.length === 1 && SNIPPET_EXIT_CHARS.has(change.text)) {
+                cancelSnippetSession(editor);
+            }
+
+            const model = editor.getModel();
+            if (!model) return;
+
+            // Calculate the offset at the end of the inserted text in the new model
+            const endOffset = change.rangeOffset + change.text.length;
```
[the remark this answers is not in the document]
Comment (tnaum-ms) [reply src/webviews/documentdb/collectionView/components/queryEditor/QueryEditor.tsx:183]: Acceptable, won't fix. Multi-cursor edits are extremely rare in a small query editor. The worst outcome is a missed auto-trigger (user can still invoke completions manually), and the snippet-exit guard already filters on `change.text.length === 1`.
